### ISSUE TITLE
Settings should not be case sensitive

### ISSUE THREAD

**Issue by jad19925** [nokia/RED#107]
In a robot file that had previously created using a basic text editor, Default tags were created in the Settings section, but the Settings editing tab in RED does not recognize them because the t in tags is lowercase.

> Default tags = unhappy RED

> Default Tags = happy RED

It would be nice if the settings tab was case insensitive as Robot itself is.

**Comment by KrzysztofJozefowicz** [comment]
thanks for issue,we are following exact names from [robot user guide](http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#id340 ) although if robot does not report any issue with case sensitivity,RED should also ignore it.
It should not be difficult to fix it (recognize without case) although other cases such as "d o cumenta t ion" should be reported as current (as error)

**Comment by adam-wypych** [comment]
I've been checked and this is not issue with recognize `Default Tags` in `RobotFrameworkCore model`. You can simple check it by adding JUnit test case to org.rf.ide.core.testdata.RobotParserTest like below:
```java
	   @Test
	    public void defaultTagsSettingCaseInsensitiveTest() {
	        // prepare
	        final String fileContent = "*** Settings ***\nDefault tags\n\n*** Test Cases ***";

	        final RobotRuntimeEnvironment runtime = mock(RobotRuntimeEnvironment.class);
	        when(runtime.getVersion()).thenReturn("2.9");
	        final RobotProjectHolder projectHolder = spy(RobotProjectHolder.class);
	        when(projectHolder.getRobotRuntime()).thenReturn(runtime);

	        // execute
	        final RobotParser parser = RobotParser.create(projectHolder, RobotParserConfig.allImportsLazy());
	        final RobotFileOutput editorContent = parser.parseEditorContent(fileContent, new File("f.robot"));

	        // verify
	        final RobotFile fileModel = editorContent.getFileModel();
	        SettingTable settingTable = fileModel.getSettingTable();
	        List<DefaultTags> defaultTags = settingTable.getDefaultTags();
	        assertThat(defaultTags).hasSize(1);
	        assertThat(defaultTags.get(0).getDeclaration().getText()).isEqualTo("Default tags");
	        assertThat(defaultTags.get(0).getDeclaration().getTypes()).containsOnly(RobotTokenType.SETTING_DEFAULT_TAGS_DECLARATION);
	        final List<RobotLine> lineContents = fileModel.getFileContent();
	        assertThat(lineContents).hasSize(4);
	        assertLine(lineContents.get(0), Arrays.asList("*** Settings ***"), "\n");
	        assertLine(lineContents.get(1), Arrays.asList("Default tags"), "\n");
	        assertLine(lineContents.get(2), Arrays.asList(), "\n");
	        assertLine(lineContents.get(3), Arrays.asList("*** Test Cases ***"), null);
	    }
```

Test will pass.

The problem is connected to class responsible for _General Settings_ extraction: org.robotframework.ide.eclipse.main.plugin.tableeditor.settings.GeneralSettingsModel. As far as I see still string representation is used instead of check `IRobotTokenType`:

```java
public class GeneralSettingsModel {

    private static final String DEFAULT_TAGS = "Default Tags";
    private static final String FORCE_TAGS = "Force Tags";
    private static final String TEST_TIMEOUT = "Test Timeout";
    private static final String TEST_TEMPLATE = "Test Template";
    private static final String TEST_TEARDOWN = "Test Teardown";
    private static final String TEST_SETUP = "Test Setup";
    private static final String SUITE_TEARDOWN = "Suite Teardown";
    private static final String SUITE_SETUP = "Suite Setup";
    
    public static List<RobotElement> findGeneralSettingsList(final RobotSettingsSection section) {
        if (section == null) {
            return newArrayList();
        }
        return newArrayList(
                Iterables.filter(newArrayList(fillSettingsMapping(section).values()),
                Predicates.notNull()));
    }

    public static Map<String, RobotElement> fillSettingsMapping(final RobotSettingsSection section) {
        final Map<String, RobotElement> initialMapping = AccessibleSettings.forFile(section.getSuiteFile())
                .createInitialMapping();
        
        if (section != null) {
            for (final RobotKeywordCall setting : section.getChildren()) {
                final String settingName = setting.getName();
                if (initialMapping.containsKey(settingName)) { // BUG HERE !!!
                    initialMapping.put(settingName, setting);
                }
            }
        }
        return initialMapping;
    }
```

**Comment by KrzysztofJozefowicz** [comment]
released in 0.7.8
